src/compiler/mir/transform/module.rs

```diff
@@ -28,18 +28,35 @@ pub fn transform(
         }
     }
 
-    let funcs = module.get_functions();
-
-    for f in funcs {
-        match f {
-            crate::compiler::ast::Item::Routine(r) => {
-                let ft = FuncTransformer::new(r.context().canonical_path(), project);
-                let p = ft.transform(r);
-                project.add_func(p)?;
+    let funcs: Vec<_> = module
+        .get_functions()
+        .iter()
+        .filter_map(|f| {
+            if let Item::Routine(r) = f {
+                Some(r)
+            } else {
+                None
             }
-            crate::compiler::ast::Item::Struct(_) => todo!(),
-            crate::compiler::ast::Item::Extern(_) => todo!(),
-        }
+        })
+        .collect();
```

Comment:
Really think that this points to a design flaw in `Module`, `get_functions` should return `RoutineDef` not `Item`.